### DISCUSSION TITLE
`into bits` format for datetimes – is %c right?

### DISCUSSION THREAD

**Discussion by robjwells** [nushell/nushell#12268] [General]
Hello, I was looking at the `into bits` command, which produces a bit string from various datatypes.

The bitstring output for a datetime value is the result of formatting the date with `%c` ([relevant line](https://github.com/nushell/nushell/blob/ff41cf91ef06c0bc3fbb8f61ec5ddd3dd175fef7/crates/nu-cmd-extra/src/extra/bits/into.rs#L210)), the "national representation of time and date".

As this will change depending on the user's system settings, and doesn't necessary match the displayed output from `date now`, would it be preferable instead to use something like the ISO date string?

An example of the current behaviour:
```nushell
〉date now
Sat, 23 Mar 2024 14:25:42 +0000 (now)
〉date now | into bits | python3 from_bits.py
Sat Mar 23 14:25:45 2024
```

(`from_bits.py` just transforms the incoming bit string into ASCII characters.)

**Comment by sholderbach** [comment]
IMO it shouldn't do this string conversion:
- either we disallow `date` as a type entirely (My preference)
  - https://github.com/nushell/nushell/pull/12313
- or expose the binary representation as unix time + offset if we are absolutely sure this will never change (which I am not so sure about, with all the trouble we had with datetime)

(repeated comment from discord)
